lib/OpenQA/WebAPI/Auth/OpenID.pm

```diff
@@ -108,8 +108,10 @@ sub auth_response ($c) {
     };
 
     $csr->handle_server_response(
-        not_openid =>
-          sub () { $err_handler->('Failed to login', 'OpenID provider returned invalid data. Please retry again') },
+        not_openid => sub () {
+            my $txt = 'Please retry again. Likely, OpenID message is not considered a valid object. It is missing, at least, "openqa.mode" value';
+            log_warning("Invalid OpenID provided. $txt");
```

Comment:
Before `$err_handler` was called, which calls `$c->flash(error => "$err: $txt")`.
I'm not sure if that's something the user would see.
But if it is, now the user wouldn't even see an error anymore?

Reply:
I dont know as well. But I guess it is fine to add this line to be consistent?

Reply:
I think the flash message is something the user would see in certain cases. (It probably depends on the page the user is getting back to.)

I'd probably try to keep the flash message, and maybe also as an error - because this is in fact why the login attempt failed.

Reply:
I added `flash` on 06f453ea6

Reply:
But I cant see any popup for warn or even debug. So I reverted it

Reply:
I would say the flash message should stay at error-level. As I said, for the single request this is a fatal error. Just from the point of view of the server admin it is nothing one can do much about so a warning makes more sense.

Reply:
> I would say the flash message should be kept and stay at error-level. As I said, for the single request this is a fatal error. Just from the point of view of the server admin it is nothing one can do much about so a warning makes more sense.

Do you ask for a change or is it just a confirmation statement?

Reply:
I ask for adding the flash message back (as error, while logging only a warning/info message on the server logs). That's at least what I think is best, not sure what the rest of the team thinks.

Reply:
done c6cc29a9b